src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -2502,6 +2535,48 @@ private static void RemoveLastNullCompletionResult(List<CompletionResult> result
             }
         }
 
+        private static void NativeCompletionScope(CompletionContext context, List<CompletionResult> result)
+        {
+            var wordToComplete = context.WordToComplete;
+
+            var typeResults = CompleteScope(wordToComplete, context);
+            if (typeResults != null)
+            {
+                result.AddRange(typeResults);
+            }
+
+            result.Add(CompletionResult.Null);
+        }
+
+        internal static List<CompletionResult> CompleteScope(string wordToComplete, CompletionContext context)
+        {
+            var results = new List<CompletionResult>();
+            var scopes = new List<string>() {"Global", "Local", "Script"};
+
+            wordToComplete = wordToComplete.ToLower();
+
+            // handle named scopes in the list above
+            foreach (var scopeName in scopes)
+            {
+                if ((scopeName.ToLower()).StartsWith(wordToComplete)) {
+                    results.Add(new CompletionResult(scopeName, scopeName, CompletionResultType.ParameterValue, "named scope"));
+                }
+            }
+
+            // handle the numbered scopes
+            for (int i = 0; i < 50; i++) {
```

Comment:
Personally, I feel it makes no sense to have the numerics.

Reply:
@theJasonHelmick - do you have an opinion on that?

Reply:
I believe there could be some use cases where this is useful. For example, if a user is in few nested scopes and they would like to see exactly how many scopes are available (possibly to reference a variable defined in a higher scope), they can cycle through the available tab completion values to confirm. In addition, when I was working on this change, I saw that there was no straightforward way to get the current scope that a user is in. With this new addition, the process to do so just involves calling the CompleteInput method.

Reply:
I generally have doubts that it is good practice to use these numbers. We should not encourage it.

Reply:
@iSazonov is there something specific in the scenarios that don't resonate with you? In some cases, you need to traverse the scopes, and it seems that this would be _extremely_ useful.

Reply:
I doubt anyone can jump from level 49 to level 7 and not make a mistake. This is also reminiscent of "goto" practice, which is not good practice. Any scriptblock can be called at an unpredictable level and an absolute reference is not reliable. Finally, I doubt anyone designs scripts with an scope level of up to 50,  but if they do, they certainly don't need this feature.

Reply:
@daxian-dbw and @adityapatwardhan - just wanted to bump this post and see if there were any other opinions on this issue.